ui/src/index.tsx

```diff
@@ -78,8 +78,15 @@ const browserHistory = useRouterHistory(createHistory)({
 const store = configureStore(loadLocalStorage(errorsQueue), browserHistory)
 const {dispatch} = store
 
-browserHistory.listen(() => {
-  dispatch(disablePresentationMode())
+// pathname of last location change
+let lastPathname: Pathname
+
+browserHistory.listen(location => {
+  // disable presentation mode only if pathname changes, #5382
+  if (lastPathname !== location.pathname) {
+    dispatch(disablePresentationMode())
+  }
+  lastPathname = location.pathname
```

Comment:
Move this line up a line into the `if` block to prevent wasted work to re-assign the same value.

Reply:
i would argue more important than not wasting cpu cycles is to make the intent clear that this should only be overwritten if they aren't equal.

Reply:
Sorry, guys, for this stupid omission. Fixed.